platforms/core-runtime/functional/src/main/java/org/gradle/internal/Combiners.java

```diff
@@ -35,4 +33,8 @@ public abstract class Combiners {
     public static <T> BinaryOperator<T> nonCombining() {
         return (BinaryOperator<T>) NON_COMBINING;
     }
+
+    private static Object apply(Object a, Object b) {
```

Comment:
:x: `apply` doesn't look like a descriptive name. Maybe something like `unsupportedCombine`? Otherwise we can just use the lambda at the constant declaration site and suppress the warning, with a justification.